src/test/java/lambda/part3/exercise/Exercise2.java

```diff
@@ -0,0 +1,161 @@
+    private static class MapHelper<T> {
+
+        private final List<T> source;
+
+        private MapHelper(List<T> source) {
+            this.source = source;
+        }
+
+        public static <T> MapHelper<T> from(List<T> source) {
+            return new MapHelper<>(source);
+        }
+
+        public List<T> getMapped() {
+            return new ArrayList<>(source);
+        }
+
+        /**
+         * Создает объект для маппинга, передавая ему новый список, построенный на основе исходного.
+         * Для добавления в новый список каждый элемент преобразовывается с использованием заданной функции.
+         * ([T], (T -> R)) -> [R]
+         *
+         * @param mapping Функция преобразования элементов.
+         */
+        public <R> MapHelper<R> map(Function<T, R> mapping) {
+            List<R> newList = new ArrayList<>();
+            for (T t : source) {
+                newList.add(mapping.apply(t));
+            }
+            return from(newList);
+        }
+
+        /**
+         * Создает объект для маппинга, передавая ему новый список, построенный на основе исходного.
+         * Для добавления в новый список каждый элемент преобразовывается в список с использованием заданной функции.
+         * ([T], (T -> [R])) -> [R]
+         *
+         * @param flatMapping Функция преобразования элементов.
+         */
+        public <R> MapHelper<R> flatMap(Function<T, List<R>> flatMapping) {
+            List<R> newList = new ArrayList<>();
+            for (T t : source) {
+                newList.addAll(flatMapping.apply(t));
+            }
+            return from(newList);
+        }
+    }
+
+    @Test
+    public void mapEmployeesToLengthOfTheirFullNamesUsingMapHelper() {
+        List<Employee> employees = getEmployees();
+
+        List<Integer> lengths = MapHelper.from(employees)
+                .map(Employee::getPerson)
+                .map(Person::getFullName)
+                .map(String::length)
+                .getMapped();
+        // TODO                 MapHelper.from(employees)
+        // TODO                          .map(Employee -> Person)
+        // TODO                          .map(Person -> String(full name))
+        // TODO                          .map(String -> Integer(length of string))
+        // TODO                          .getMapped();
+        assertEquals(Arrays.asList(14, 19, 14, 15, 14, 16), lengths);
+    }
+
+    @Test
+    public void mapEmployeesToCodesOfLetterTheirPositionsUsingMapHelper() {
+        List<Employee> employees = getEmployees();
+
+        List<Integer> codes = MapHelper.from(employees)
+                .flatMap(Employee::getJobHistory)
+                .map(JobHistoryEntry::getPosition)
+                .flatMap(s -> {
+                    List<Character> chars = new ArrayList<>();
+                    for(Character letter : s.toCharArray()){
+                        chars.add(letter);
+                    }
+                    return chars;
+                })
```

Comment:
It's better for code clarity to replace statement lambda with method reference or at least to ```Function<>``` variable